src/Sarif/ExtensionMethods.cs

```diff
@@ -16,6 +16,40 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        public static string InstanceIdInstanceComponent(this RunAutomationDetails runAutomationDetails)
+        {
+            string instanceId = runAutomationDetails.InstanceId;
+
+            if (instanceId == null)
+            {
+                return null;
+            }
+
+            if (instanceId.EndsWith("/"))
+            {
+                return String.Empty;
+            }
+
+            return instanceId.Substring(instanceId.LastIndexOf('/') + 1);
+        }
+
+        public static string InstanceIdLogicalComponent(this RunAutomationDetails runAutomationDetails)
+        {
+            string instanceId = runAutomationDetails.InstanceId;
+
+            if (instanceId == null)
+            {
+                return null;
+            }
+
+            if (!instanceId.Contains("/"))
+            {
+                return String.Empty;
+            }
+
+            return instanceId.Substring(0, instanceId.LastIndexOf('/') + 1);
```

Comment:
>+ 1 [](start = 71, length = 3)

This includes the final slash. I don't agree with that. If `instanceId` is `x/y`, then the "category" (the logical id) is `x`, not `x/`. I know you love your trailing slashes on directory-valued URIs, but I don't think it's right here.

Reply:
(Reactivated to call my response to your attention.)

---
In reply to: [226872911](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226872911) [](ancestors = 226872911,226844096,226712702)